opensrp-chw/src/main/java/org/smartregister/chw/fragment/FamilyRegisterFragment.java

```diff
@@ -242,6 +245,74 @@ private String dueFilterAndSortQuery() {
         return query;
     }
 
+    private String defaultFilterAndSortQuery() {
+        SmartRegisterQueryBuilder sqb = new SmartRegisterQueryBuilder(mainSelect);
+
+        String query = "";
+        try {
+            if (isValidFilterForFts(commonRepository())) {
+
+                StringBuilder sb = new StringBuilder();
+
+                if (ArrayUtils.isEmpty(joinTables)) {
+
+                    sb.append(MessageFormat.format("SELECT {0} FROM {1} ", CommonFtsObject.idColumn, CommonFtsObject.searchTableName(tablename)));
+                    if (StringUtils.isNotBlank(mainCondition)) {
+                        sb.append(MessageFormat.format("WHERE {0} ", mainCondition));
+                    }
+
+                    if (StringUtils.isNotBlank(filters)) {
+                        sb.append(MessageFormat.format("AND PHRASE MATCH '{0}*' ", filters));
+                    }
+
+                    sb.append(MessageFormat.format("LIMIT {0} , {1} ", clientAdapter.getCurrentoffset(), clientAdapter.getCurrentlimit()));
+
+                } else {
+
+                    StringBuilder sbJoin = new StringBuilder();
+                    for (String tableName : joinTables) {
+
+                        if (StringUtils.isNotBlank(sbJoin.toString().trim())) {
+                            sbJoin.append("UNION ");
+                        }
+
+                        sbJoin.append(MessageFormat.format("SELECT {0} FROM {1} WHERE date_removed is null ", CommonFtsObject.relationalIdColumn, CommonFtsObject.searchTableName(tableName)));
```

Comment:
Change "date_removed is null" to mainCondition variable which may change

Reply:
Updated this. Refactored the query to use a more test friendly design